scripts/airgap.sh

```diff
@@ -0,0 +1,195 @@
+      shift 2;;
+   -s|--kwctl-sources)
+      KWCTL_SOURCES="$2"
+      shift 2;;
+   -h|--help)
+      usage
+      exit 0;;
+   *)
+      usage
+      exit 1;;
+   esac
+done
+set -- "${POSITIONAL_ARGS[@]}" # restore positional parameters
+
+trap 'echo "Error on ${BASH_SOURCE}:${LINENO} $(sed -n "${LINENO} s/^\s*//p" ${BASH_SOURCE})"' ERR
+
+# ==================================================================================================
+# Check parameters, requirements and set defaults
+
+# TODO - delme
+: ${KWCTL_SOURCES:=sources.yaml}
+: ${REGISTRY:=localhost:5000}
+
+# Check parameters & Requirement
+[[ "${action:-}" =~ ^(pull|push|setup)$ ]] || { echo "Action not recognized: ${action:--}"; exit 1; }
+[[ "$action" =~ ^(push|setup)$ ]] && : ${REGISTRY:?Required parameter is missing: --registry}
+[ -v REGISTRY ] && curl --output /dev/null --silent --head --fail "$REGISTRY" # check registry is online
+
+[ -v IMAGES_TXT ] && test -f "$IMAGES_TXT"
+[ -v POLICIES_TXT ] && test -f "$POLICIES_TXT"
+[ -v KWCTL_SOURCES ] && test -f "$KWCTL_SOURCES"
+if [ "$action" == "push" ]; then
+   [ -v IMAGES_TAR ] && test -f "$IMAGES_TAR"
+   [ -v POLICIES_TAR ] && test -f "$POLICIES_TAR"
+fi
+which jq kwctl helm docker > /dev/null || { echo "Missing one of requirements"; exit 1; }
+
+# Set defaults
+: ${IMAGES_TAR:=kubewarden-images.tar}
+: ${POLICIES_TAR:=kubewarden-policies.tar.gz}
+
+kwctl_args=${KWCTL_SOURCES:+--sources-path $KWCTL_SOURCES}
+tmpfile=$(mktemp /tmp/kubewarden-airgap.XXXXXX)
+
+# ==================================================================================================
+# Functions
+
+# Print green colored message
+function step() { printf -- "\e[32m${*}\e[0m\n"; }
+
+# Get latest policiy and image lists from artifacthub
+# Use text files from kubewarden-defaults chart
+function pull_latest_lists() {
+   tag=$(helm search hub kubewarden-defaults -o json | jq -er '.[] | select(.repository.url=="https://charts.kubewarden.io").version')
+   assets_urls=$(curl -sL https://api.github.com/repos/kubewarden/helm-charts/releases/tags/kubewarden-defaults-$tag | jq -er '.assets[].browser_download_url')
+   imagelist_url=$(grep 'images.txt$' <<< "$assets_urls")
+   policylist_url=$(grep 'policylist.txt$' <<< "$assets_urls")
+}
+
+# Download images & policies
+# If no $source is set use defaults from pull_latest_lists
+function pull() {
+   local source="$1"    # file or url with list of images or policies
+   local target="$2"    # target archive filename
+   local command        # docker for images or kwctl for policies
+
+   [ -f "$target" ] && { echo "File $target already exists, skipping pull."; return; }
+
+   # if source is url download it
+   if [[ "$source" =~ ^https:// ]]; then
+      curl -sL $source -o "$tmpfile"
+      source=$tmpfile
+   fi
+
+   grep -qF '://' $source && command=kwctl || command=docker
+   cat $source | while read line || [[ -n $line ]]; do
+      echo "Pull: $line"
+      $command pull "$line" > /dev/null
```

Comment:
```suggestion
      $command pull "$line"
```

I am happy seeing the list of things being pulled; ideally the script is run only once, can be verbose.